addons/udes_stock/tests/test_stock_move_line.py

```diff
@@ -540,6 +540,511 @@ def test_over_mark_error(self):
         )
         self.assertEqual(e.exception.args[0], msg)
 
+    def test_swap_serial_lot_name_on_single_move_line(self):
+        """
+        When prepare is given a different lot_name for a single serial product move_line, and u_tracked_product_swap is turned on, then the resultant move_line values
+        will have swapped lot_names
+        """
+        self.picking_type_pick.u_allow_swapping_tracked_products = True
+        for lot_num in range(1, 6):
+            self.create_quant(
+                self.strawberry.id,
+                location_id=self.test_stock_location_01.id,
+                qty=1,
+                lot_name=f"{lot_num}",
+            )
+
+        product_info = [{"product": self.strawberry, "uom_qty": 1}]
+        picking = self.create_picking(
+            self.picking_type_pick, products_info=product_info, assign=True
+        )
+
+        product_ids = [{"barcode": "productStrawberry", "uom_qty": 1, "lot_names": ["2"]}]
+        location = self.test_stock_location_01
+        pallet_to_pick_onto = self.create_package(name="UDES1")
+
+        res = picking.move_line_ids.prepare(
+            product_ids=product_ids, location=location, result_package=pallet_to_pick_onto
+        )
+        self.assertEqual(res[picking.move_line_ids]["lot_name"], "2")
+        self.assertEqual(res[picking.move_line_ids]["product_uom_qty"], 1)
+        self.assertEqual(res[picking.move_line_ids]["qty_done"], 1)
+        for mls, mls_values in res.items():
+            mls.mark_as_done(mls_values)
+        self.assertEqual(picking.move_line_ids.lot_id.mapped("name"), ["2"])
+
+    def test_swap_serial_lot_names_on_multiple_move_lines(self):
+        """
+        When prepare is given different lot_names for multiple serial product move_lines, and u_tracked_product_swap is turned on, then the resultant move_line values
+        will have swapped lot_names
+        """
+        self.picking_type_pick.u_allow_swapping_tracked_products = True
+        for lot_num in range(1, 6):
+            self.create_quant(
+                self.strawberry.id,
+                location_id=self.test_stock_location_01.id,
+                qty=1,
+                lot_name=f"{lot_num}",
+            )
+        product_info = [{"product": self.strawberry, "uom_qty": 3}]
+        picking = self.create_picking(
+            self.picking_type_pick, products_info=product_info, assign=True
+        )
+
+        product_ids = [{"barcode": "productStrawberry", "uom_qty": 3, "lot_names": ["1", "4", "5"]}]
+        location = self.test_stock_location_01
+        pallet_to_pick_onto = self.create_package(name="UDES1")
+
+        res = picking.move_line_ids.prepare(
+            product_ids=product_ids, location=location, result_package=pallet_to_pick_onto
+        )
+        for mls, mls_values in res.items():
+            mls.mark_as_done(mls_values)
+        self.assertIn(picking.move_line_ids.lot_id.mapped("name")[0], ["1", "5", "4"])
```

Comment:
What about only one assert equatl with the lto names sorted? We want to make sure we exactly have all three.